DOC-1842 K8s AA preview edit participating clusters (#2488)
RedisLabs/redislabs-docs: content/kubernetes/active-active/preview/edit-clusters.md

See [Create Active-Active databases for Kubernetes]({{<relref "/kubernetes/active-active/create-aa-database.md">}}) for the currently supported procedure.
{{</banner-article>}}

## Add a participating cluster

Use the following steps to add a participating cluster to an existing Redis Enterprise Active-Active database (REAADB) for Kubernetes.

### Prerequisites

To prepare the Redis Enterprise cluster (REC) to participate in an Active-Active database, perform the following tasks from [Prepare participating clusters]({{<relref "/kubernetes/active-active/preview/prepare-clusters.md">}}):

- Make sure the cluster meets the hardware and naming requirements. 
- Enable the Active-Active controllers.
- Configure external routing.
- Configure `ValidatingWebhookConfiguration`.

### Collect REC credentials

To communicate with other clusters, all participating clusters need access to the admin credentials for all other clusters.

1. Locate the file holding the admin credentials for all participating RECs created while [preparing the clusters]({{<relref "/kubernetes/active-active/preview/prepare-clusters.md">}}) (such as `all-rec-secrets.yaml`).

> 1. Within that file, create a secret for the new participating cluster named `redis-enterprise-<rerc>`. 

Reviewer: I think the step numbering is wrong.
this should be 2. and etc...

  {{<note>}}The file should contain a secret for each existing participating cluster.{{</note>}}

    The example below shows a file (`all-rec-secrets.yaml`) holding secrets for two participating clusters:

    ```yaml
    apiVersion: v1
    data:
      password: 
      username: 
    kind: Secret
    metadata:
      name: redis-enterprise-rerc1
    type: Opaque

    ---

    apiVersion: v1
    data:
      password: 
      username: 
    kind: Secret
    metadata:
      name: redis-enterprise-rerc2
    type: Opaque

    ```

1. Get the REC credentials secret for the new participating cluster.

    ```sh
    kubectl get secret -o yaml <rec-name>
    ```

    The admin credentials secret for an REC named `rec1` has this format:

    ```yaml
    apiVersion: v1
    data:
      password: ABcdef12345
      username: GHij56789
    kind: Secret
    metadata:
      name: rec1
    type: Opaque
    ```

> 1. Add the username and password to the new secret for that REC and namespace.

Reviewer: the name is no longer with the rec name and ns convention thus please change this statement to:
Add the username and password to the new secret for that new participating cluster.
note: the secret name convention is redis-enterprise-<RERC name> where RERC name is the corresponding RERC custom resource for that participating cluster.

>     This example shows the collected secrets file (`all-rec-secrets.yaml`) for `rec1` in namespace `ns1` and `rec2` in namespace `ns2`.

Reviewer: also here it is no longer the convention of the rec name and ns.
suggestion:
This example shows the collected secrets file (`all-rec-secrets.yaml`) for `rerc1` that represents the `rec1` in namespace `ns1`, `rerc2` that represents the `rec2` in namespace `ns2` and `rerc3` that represents the `rec3` in namespace `ns3.

note that I have added rerc3 since we start here (in the previous steps) with example that shows 2 secrets we now add the 3rd to stay consisted.

    ```yaml
    apiVersion: v1
    data:
      password: ABcdef12345
      username: GHij56789
    kind: Secret
    metadata:
      name: redis-enterprise-rerc1
    type: Opaque

    ---

    apiVersion: v1
    data:
      password: KLmndo123456
      username: PQrst789010
    kind: Secret
    metadata:
      name: redis-enterprise-rerc2
    type: Opaque

> 

Reviewer: please add here 'rerc3' to stay consisted with the previous steps as we showed before example with 2 participating clusters secrets and now here we add the 3rd.
this is the the 3rd rerc should be like:
"""
---
    apiVersion: v1
    data:
      password: KLmndo123456
      username: PQrst789010
    kind: Secret
    metadata:
      name: redis-enterprise-rerc3
    type: Opaque
"""

    ```sh
    kubectl apply -f <all-rec-secrets-file>
    ```

 If the admin credentials for any of the clusters change, update and reapply the file to all clusters.

### Create RERC

1. From one of the existing participating clusters, create a `RedisEnterpriseRemoteCluster` (RERC) custom resource file for the new participating cluster. 

  This example shows a RERC custom resource for an REC named `rec3` in the namespace `ns3`. 

  ```yaml
  apiVersion: app.redislabs.com/v1alpha1
  kind: RedisEnterpriseRemoteCluster
  metadata:
    name: rerc3
  spec:
    recName: rec3
    recNamespace: ns3
    apiFqdnUrl: test-example-api-rec3-ns3.redis.com
    dbFqdnSuffix: -example-cluster-rec3-ns3.redis.com
    secretName: redis-enterprise-rerc3
  ```

1. Create the RERC custom resource. 

  ```sh
  kubectl create -f <new-RERC-file>
  ```

1. Check the status of the newly created RERC custom resource.

  ```sh
  kubectl get rerc <RERC-name>
  ```

  The output should look like this: 
  ```sh
    NAME        STATUS   SPEC STATUS   LOCAL
  rerc3   Active   Valid         true
  ```

### Edit REAADB spec

1. Add the new RERC name to the `participatingClusters` list in the REAADB spec.

  ```sh

>   kubectl patch reaadb < --type merge --patch '{"spec": {"participatingClusters": [{"name": "rerc3"}]}}'

Reviewer: maybe add a note that this patch is adding the RERC custom resource named `rerc3`, so it will be clear that the customer should edit the kubectl command.

Reviewer: also, this command is broken
it should be:
kubectl patch reaadb example-aadb-1 --type merge --patch '{"spec": {"participatingClusters": [{"name": "rerc3"}]}}'

and there should be notes that this will patch the REDB named: `example-aadb-1` with the RERC named `rerc3`

  ```

1. View the REAADB `participatingClusters` status to verify the cluster was added.

  ```sh

>   kubectl get reaadb <REAADB-name> -o=jsonpath='{.status.participatingClusters}'

Reviewer: please add the expected output with the example from the AA readme so this will be more clear

    ```sh
    kubectl get secret -o yaml <rec-name>
    ```

    The admin credentials secret for an REC named `rec1` would be similar to this:

    ```yaml
    apiVersion: v1
    data:
      password: ABcdef12345
      username: GHij56789
    kind: Secret
    metadata:
      name: rec1
    type: Opaque
    ```

1. Add the username and password to the new secret for that REC and namespace.

    This example shows the collected secrets file (`all-rec-secrets.yaml`) for `rec1` in namespace `ns1` and `rec2` in namespace `ns2`.

    ```yaml
    apiVersion: v1
    data:
      password: ABcdef12345
      username: GHij56789
    kind: Secret
    metadata:
      name: redis-enterprise-rerc1
    type: Opaque

    ---

    apiVersion: v1
    data:
      password: KLmndo123456
      username: PQrst789010
    kind: Secret
    metadata:
      name: redis-enterprise-rerc2
    type: Opaque

    ```

1. Apply the file of collected secrets to every participating REC.

    ```sh
    kubectl apply -f <all-rec-secrets-file>
    ```

 If the admin credentials for any of the clusters changes, the file will need to be updated and reapplied to all clusters.

### Create RERC

> 1. From one of the exiting particpating clusters, create a `RedisEnterpriseRemoteCluster` (RERC) custom resource file for the new participating cluster. 

Reviewer 2: ```suggestion
1. From one of the existing participating clusters, create a `RedisEnterpriseRemoteCluster` (RERC) custom resource file for the new participating cluster. 
```

  This example shows a RERC custom resource for an REC named `rec3` in the namespace `ns3`. 

  ```yaml
  apiVersion: app.redislabs.com/v1alpha1
kind: RedisEnterpriseRemoteCluster
metadata:
  name: rerc3
spec:
  recName: rec3
  recNamespace: ns3
  apiFqdnUrl: test-example-api-rec3-ns3.redis.com

>   dbFqdnSuffix: -example-cluster-rec3-ns3.redis.com

Reviewer 2: ```suggestion
    dbFqdnSuffix: -example-cluster-rec3-ns3.redis.com
```

  secretName: redis-enterprise-rerc3
  ```

1. Create the RERC custom resource. 

  ```sh
  kubectl create -f <new-RERC-file>
  ```

1. Check the status of the newly created RERC custom resource.

  ```sh
  kubectl get rerc <RERC-name>
  ```

  The output should look like this: 
  ```sh
    NAME        STATUS   SPEC STATUS   LOCAL
  rerc3   Active   Valid         true
  ```

> 

Reviewer 2: ```suggestion
```
